ttnn/cpp/ttnn/operations/normalization/layernorm/layernorm.hpp

```diff
@@ -0,0 +1,130 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "tt_eager/tt_dnn/op_library/layernorm/layernorm_op.hpp"
+
+namespace ttnn {
+namespace operations {
+namespace normalization {
+
+struct LayerNorm {
+    static inline const std::array<ttnn::TensorSchema, 4> input_tensor_schemas() {
+        return {
+            ttnn::TensorSchema{
+                2,
+                4,
+                {ttnn::bfloat16, ttnn::bfloat8_b, ttnn::bfloat4_b},
+                {ttnn::TILE_LAYOUT},
+                true,
+                false,
+                false,
+                false},
+            ttnn::TensorSchema{
+                1,
+                4,
+                {ttnn::bfloat16, ttnn::bfloat8_b, ttnn::bfloat4_b},
+                {ttnn::TILE_LAYOUT, ttnn::ROW_MAJOR_LAYOUT},
+                true,
+                false,
+                false,
+                true},
+            ttnn::TensorSchema{
+                1,
+                4,
+                {ttnn::bfloat16, ttnn::bfloat8_b, ttnn::bfloat4_b},
+                {ttnn::TILE_LAYOUT, ttnn::ROW_MAJOR_LAYOUT},
+                true,
+                false,
+                false,
+                true},
+            ttnn::TensorSchema{
+                1,
+                4,
+                {ttnn::bfloat16, ttnn::bfloat8_b, ttnn::bfloat4_b},
+                {ttnn::TILE_LAYOUT, ttnn::ROW_MAJOR_LAYOUT},
+                true,
+                false,
+                false,
+                true}};
+    }
+
+    template <typename... Args>
+    static auto input_tensors_to_validate(
+        const Tensor& input_tensor,
+        float epsilon = 1e-12,
+        const std::optional<const ttnn::Tensor>& weight = std::nullopt,
+        const std::optional<const ttnn::Tensor>& bias = std::nullopt,
+        const std::optional<const ttnn::Tensor>& residual_input_tensor = std::nullopt,
+        Args&&... args) {
+        return std::forward_as_tuple(input_tensor, weight, bias, residual_input_tensor);
+    }
+
+    static inline ttnn::Tensor execute_on_worker_thread(
+        const ttnn::Tensor& input_tensor,
+        float epsilon = 1e-12,
+        const std::optional<const ttnn::Tensor>& weight = std::nullopt,
+        const std::optional<const ttnn::Tensor>& bias = std::nullopt,
+        const std::optional<const ttnn::Tensor>& residual_input_tensor = std::nullopt,
+        const std::optional<MemoryConfig>& memory_config_arg = std::nullopt,
```

Comment:
Please keep it more consistent. Other args don't have _arg, and group_norm op doesn't have _arg in memory_config.

Reply:
I'll make the change in layernorm PR, this is just copied code from normalization.hpp, which is already in ttnn.